pkg/orchestration/wiring/kubeingress/plugin.go

```diff
@@ -232,19 +266,38 @@ func buildIngressHostName(resourceName voyager.ResourceName, sc wiringplugin.Sta
 		clusterHostPath)
 }
 
+func extractSingleDependencyOfType(dependencies []wiringplugin.WiredDependency, resourceType voyager.ResourceType) (*wiringplugin.WiredDependency, bool /* retriable */, error) {
+	var matchedDependency *wiringplugin.WiredDependency
+	for x, dependency := range dependencies {
+		if dependency.Type == resourceType {
+			if matchedDependency != nil {
+				return nil, false, errors.Errorf("must depend on a single %s resource, but multiple were found", resourceType)
+			}
+			matchedDependency = &dependencies[x]
+		}
+	}
+
+	if matchedDependency == nil {
+		return nil, false, errors.Errorf("must depend on a single %s resource, but none were found", resourceType)
+	}
+
+	return matchedDependency, false, nil
+}
+
 func extractKubeComputeDependency(dependencies []wiringplugin.WiredDependency) (*apps_v1.Deployment, bool /* retriable */, error) {
 	// Require exactly one KubeCompute dependency
-	if len(dependencies) != 1 || dependencies[0].Type != apik8scompute.ResourceType {
-		return nil, false, errors.Errorf("must depend on a single %s resource. %d dependencies were given", apik8scompute.ResourceType, len(dependencies))
+	kubeComputeDependency, retriable, err := extractSingleDependencyOfType(dependencies, apik8scompute.ResourceType)
```

Comment:
I didn't realize earlier you are changing the logic here. What is the reason?
We can leave the code as is or we can accept your version - it does not matter because we are refactoring this in autowiring2 (`SmithResources` is going away).

Reply:
I didn't see any reason the ingress shouldn't be allowed to depend on any arbitrary resource, so long as there's exactly one kubecompute dependency.

Reply:
But what does it mean to `depend on any arbitrary resource`? Why a user might want to do it? The reason to only allow to depend on certain resource types is to prevent user mistakes. They might assume that dependency actually does something when it does not. Or they may have wanted to depend on some other object and used a wrong name by mistake. System will not let them know.